docs/how-to/h-upgrade.md

```diff
@@ -0,0 +1,115 @@
+# How to upgrade between minor versions
+
+> **:warning: WARNING:** this is currently a feature in "edge" or "beta" risks channels. We do NOT recommend to use features in production that are not in the `stable` channel! Also, contact [Canonical Data Platform team](https://chat.charmhub.io/charmhub/channels/data-platform) if you are interested in the topic.
+
+Charm upgrades allow admin to upgrade both operator code (e.g. the revision used by the charm) and/or the workload version. Note that since the charm code pin a particular version of the workload, a charm upgrade may or may not involve also a workload version upgrade. In general, the following guide only applies for in-place upgrade that involve at most minor version upgrade of Kafka workload, e.g. between Kafka 3.4.x to 3.5.x. Major workload upgrades are generally **NOT SUPPORTED** and they should be carried out using out-of-place migrations. Please refer to the how-to guide about cluster migration [here](/t/charmed-kafka-how-to-cluster-migration/10951) for more information on how this can be achieved.
+
+We strongly recommend to **NOT** perform any other extraordinary operations on the Kafka cluster, while upgrading. As an examples, these may be (but not limited) to the following:
+1. Adding or removing units
+2. Creating or destroying new relations
+3. Changes in workload configuration
+4. Upgrading other connected applications (e.g. Zookeeper)
```

Comment:
**nitpick:** ZooKeeper
**suggestion:** maybe explicitly mention it's not advised to upgrade zk 'at the same time'. But say that it might be necessary before

Reply:
It is clarified at the bottom of the page, what to do for concurrent Zk and Kafka upgrades